test/test-packages.el

```diff
@@ -34,6 +34,12 @@
   "Get version of the package by NAME."
   (let ((pkg (cadr (assq name package-alist)))) (when pkg (package-desc-version pkg))))
 
+(defun test-packages--install-from-archive (pkg-desc)
+  "Advice execute beofre function `package-install-from-archive'."
+  (setq byte-compile-error-on-warn
+        (ignore-errors (string-prefix-p "lsp-" (symbol-name (package-desc-name pkg-desc))))))
```

Comment:
Does this works for you? Or do you want to compare it with a emacs-lsp package list?

Reply:
That won't work for `helm-lsp` :/

Reply:
I can't see a easy way without adding a list of all known downstreams

Reply:
I have changed to 

```el
(advice-add
   'package-install-from-archive
   :before (lambda (pkg-desc)
             (setq byte-compile-error-on-warn
                   (ignore-errors (memq (package-desc-name pkg-desc) pkgs)))))
```

Reply:
@yyoncho I have applied it this way.